src/main/java/com/faforever/client/chat/ChatUserControl.java

```diff
@@ -76,17 +115,115 @@ public PlayerInfoBean getPlayerInfoBean() {
+        String ladderRating = i18n.get("playerRatingFormat.ladder", RatingUtil.getLadderRating(playerInfoBean));
+
+        userRatingTooltip.setText(String.format("%s \n %s", globalRating, ladderRating));
+
+        playerInfoBean.globalRatingMeanProperty().addListener((observable, oldValue, newValue) -> {
+          String updatedGlobalRating = i18n.get("playerRatingFormat.global", RatingUtil.getGlobalRating(playerInfoBean));
+          String oldLadderRating = i18n.get("playerRatingFormat.ladder", RatingUtil.getLadderRating(playerInfoBean));
+          userRatingTooltip.setText(String.format("%s \n %s", updatedGlobalRating, oldLadderRating));
+        });
+
+        playerInfoBean.ladderRatingMeanProperty().addListener((observable, oldValue, newValue) -> {
+          String oldGlobalRating = i18n.get("playerRatingFormat.global", RatingUtil.getGlobalRating(playerInfoBean));
+          String updatedLadderRating = i18n.get("playerRatingFormat.ladder", RatingUtil.getLadderRating(playerInfoBean));
+          userRatingTooltip.setText(String.format("%s \n %s", oldGlobalRating, updatedLadderRating));
+        });
+
+        Tooltip.install(clanLabel, userRatingTooltip);
+        Tooltip.install(usernameLabel, userRatingTooltip);
+      }
+    });
+
+  }
+
+  private void configureColor() {
+    ChatPrefs chatPrefs = preferencesService.getPreferences().getChatPrefs();
+    ObjectProperty<Color> colorProperty = null;
+    Color color = null;
+
+    //Always regardless of pretty Colors enabled
+    if (playerInfoBean.getModeratorInChannels().size() > 0) {
+      colorProperty = chatPrefs.modsChatColorProperty();
+    } else if (playerInfoBean.isFriend()) {
+      colorProperty = chatPrefs.friendsChatColorProperty();
+    } else if (playerInfoBean.isFoe()) {
+      colorProperty = chatPrefs.foesChatColorProperty();
+    } else if (playerInfoBean.getUsername().equals(userService.getUsername())) {
+      colorProperty = chatPrefs.selfChatColorProperty();
+    }
+
+    //Only if pretty colors is disabled
+    if (!chatPrefs.getPrettyColors() && colorProperty == null) {
+      if (playerInfoBean.isChatOnly()) {
+        colorProperty = chatPrefs.ircChatColorProperty();
+      } else {
+        colorProperty = chatPrefs.othersChatColorProperty();
+      }
+
+      //Self color
+    } else if (chatPrefs.getPrettyColors() && colorProperty == null) {
+      color = ColorGeneratorUtil.generatePrettyHexColor();
+      ChatUser chatUser = chatService.getChatUser(playerInfoBean.getUsername());
+
+      try {
+        if (chatUser.getColor() == null) {
+          chatUser.setColor(color);
+        } else {
+          color = chatUser.getColor();
+        }
+      } catch (NullPointerException e) {
+        logger.warn("Could not generate color for {}", playerInfoBean.getUsername());
+      }
+    }
+
+    if (colorProperty != null) {
+      //FIXME if there is a listener already will this break?
```

Comment:
The behavior will be undefined; depends on which listener triggers first. I would create one listener instance as a field and then first remove it from all properties before adding it to the desired one. Otherwise we'll have a lot of stale listeners (and undefined behavior anyway).